From ac9cb59cfdc222666f220c835e7f24af7e146e5a Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 01 Mar 2021 12:19:01 +0200
Subject: [PATCH] [SW-30985] gaudi2_arc_gnu: add support to bringup preboot

This patch brings up preboot, just like in the case
of greco, using the verification API call from
the boot rom.

Change-Id: Id473cc4342d030168861612d6149c6d6512723e3
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2_arc_gnu/Makefile b/gaudi2_arc_gnu/Makefile
index ed41eb2..5f55fc5 100644
--- a/gaudi2_arc_gnu/Makefile
+++ b/gaudi2_arc_gnu/Makefile
@@ -1,6 +1,8 @@
 FW_SIZE := 0x8000
 PPBOOT_MAX_SIZE := 16384
 
+LDFLAGS += -Wl,--just-symbols ../arcboot-custom/exports.elf
+
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/pll.c)
 #PREBOOT_SRCS_C += $(shell ls preboot/thermal_v2.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/security.c)
diff --git a/gaudi2_arc_gnu/common.c b/gaudi2_arc_gnu/common.c
index 41347b4..b5b8750 100644
--- a/gaudi2_arc_gnu/common.c
+++ b/gaudi2_arc_gnu/common.c
@@ -36,6 +36,7 @@
 #include "common.h"
 #include <bitops.h>
 #include "preboot/thermal.h"
+#include <arcboot_export.h>
 
 #define FW_WORD_NUM		0x2200	/* FW_SIZE / 2 */
 #define PRODUCT_ID_STR_SIZE	0x8
@@ -447,3 +448,9 @@
 
 	return PREBOOT_TEXT_BASE;
 }
+
+bool ppboot_load_preboot_img(u64 *loadaddr)
+{
+	*loadaddr = copy_preboot();
+	return true;
+}
diff --git a/gaudi2_arc_gnu/common.h b/gaudi2_arc_gnu/common.h
index 33cd632..9a9712e 100644
--- a/gaudi2_arc_gnu/common.h
+++ b/gaudi2_arc_gnu/common.h
@@ -17,7 +17,7 @@
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
 
-#define HL_SP_SRAM_BASE		0x7FF7FE0000ULL
+#define HL_SP_SRAM_BASE		0x1000007FF7FE0000ULL
 #define PCIE_FW_SRAM_BASE	0x27FF0000
 #define FW_SRAM_BASE		0x27FD0000
 #define SP_SRAM_BASE		0x27FE0000
@@ -48,13 +48,10 @@
 #define I2C_SLAVE_ADDRESS	0x6A
 #define I2C_SLAVE_ADDRESS_DBG	I2C_SLAVE_ADDRESS
 
-/* TODO: Check these */
 #define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK        BIT(19)
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI	BIT(14)
-#define EFUSE_SYSCFG_KEY_LOCATION_SPI		BIT(8)
-#define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI	BIT(6)
-
+#define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI	BIT(8)
 #define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
+
 #define RECOVERY_TEXT_BASE	(0x0 + 0x4)
 #define RECOVERY_SIZE		0x4000
 
diff --git a/ppboot/ppboot.c b/ppboot/ppboot.c
index f56ee1c..50e199b 100644
--- a/ppboot/ppboot.c
+++ b/ppboot/ppboot.c
@@ -70,7 +70,7 @@
 	return is_valid;
 }
 
-static bool ppboot_load_preboot_img(u64 *loadaddr)
+__weak bool ppboot_load_preboot_img(u64 *loadaddr)
 {
 	struct boot_desc desc;
 	bool ret;
